src/components/TextInputWithLabel.js

```diff
@@ -0,0 +1,39 @@
+import _ from 'underscore';
+import React from 'react';
+import {Text, TextInput} from 'react-native';
+import PropTypes from 'prop-types';
+import styles from '../styles/styles';
+
+const propTypes = {
+    // Label text
+    label: PropTypes.string.isRequired,
+
+    // Whether this field has an error
+    hasError: PropTypes.bool,
+
+    // Text to show if there is an error
+    errorText: PropTypes.string,
+};
+
+const defaultProps = {
+    hasError: false,
+    errorText: '',
+};
+
+const TextInputWithLabel = props => (
+    <>
+        <Text style={[styles.formLabel]}>{props.label}</Text>
+        <TextInput
+            style={[styles.textInput, styles.mb1]}
+            // eslint-disable-next-line react/jsx-props-no-spreading
+            {..._.omit(props, ['label'])}
```

Comment:
NAB & Pet peevy, but imo this is more readable:
```suggestion
            {...(_.omit(props, ['label']))}
```

Reply:
Actually, is there any reason we can't just pass the `label` prop along too? AFAIK `TextInput` doesn't take a `label` prop, so passing one in should have no effect.

Reply:
While it probably doesn't hurt to pass it, I think we should be explicit about what is passed and what isn't. It just makes the code more predictable.